apps/etl-func/src/adapters/functions/__tests__/messages-ingestion.test.ts

```
@@ -54,6 +57,9 @@ const producer = new EventHubEventProducer(
   eventHubProducerClient,
   messageSchema,
 );
+
+const eventErrorRepository = new mocks.EventErrorRepository();
+
```

I would update all the test cases in this file now that we have added a new component that impact the business logic.

- `shoud resolve if nothing throws` we should check that the `push` is not called
- `shoud call multiple times the business logic function if documents are more than one` we should check that the `push` is not called
- `shoud resolve if the documents array has malformed objects` we should check that the `publish` is called  exactly 1 time
- `should not call any function if the documents array is empty` we should check that the `publish` is not called
- `should throw an error if tokenize throws an error` we should check that the `publish` is called exactly 1 time
- `should resolve calling tokenize and getMessageByMetadata onfly for documents with isPending = false` we should check that the `publish` is not called

---

01a94e1c4205786e609900bae9d0905bc69f7b31